platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -720,9 +720,48 @@ public VirtualLeafRecord<K, V> lookupLeafByPath(final long path, final boolean f
                 : lookupLeafByKey(mutation.value, forModify);
     }
 
+    /**
+     * Returns a stream of dirty leaves from this cache instance to hash this virtual map copy.
+     *
+     * @param firstLeafPath
+     * 		The first leaf path to include to the stream
+     * @param lastLeafPath
+     *      The last leaf path to include to the stream
+     * @return
+     *      A stream of dirty leaves for hashing
+     */
+    public Stream<VirtualLeafRecord<K, V>> dirtyLeavesForHash(final long firstLeafPath, final long lastLeafPath) {
+        if (dirtyLeaves.isMerged()) {
```

Comment:
I think `isMerged()` should be defined in `VirtualNodeCache` rather than `ConcurrentArray`.

Reply:
Good idea. Let me implement it